merkle-tree/indexed/src/lib.rs

```diff
@@ -229,4 +232,35 @@ where
 
         Ok(())
     }
+
+    /// Initializes the address merkle tree with the given initial value.
+    /// The initial value should be a high value since one needs to prove non-inclusion of an address prior to insertion.
+    /// Thus, addresses with higher values than the initial value cannot be inserted.
+    pub fn initialize_address_merkle_tree(
+        address_merkle_tree_inited: &mut IndexedMerkleTree<'a, H, I, HEIGHT>,
```

Comment:
Even if we don't end up moving everything from this method to `init()`, can we change the first argument just to

```suggestion
        &mut self,
```
?

It will still work for a `&mut IndexedMerkleTree`, but also for other mutable types of `IndexedMerkleTree`.

Reply:
Didn't see the other init method before, unified them and removed the initial append.